rpcs3/rpcs3qt/main_window.cpp

```diff
@@ -189,6 +190,124 @@ QIcon main_window::GetAppIcon()
 	return m_appIcon;
 }
 
+void main_window::InstallDiscExtra(const std::string path)
+{
+	// Get title id from PARAM.sfo
+	const auto insdir_psf = psf::load_object([&]
+	{
+		return fs::file(path + "/PS3_GAME/PARAM.SFO");
+	}());
+	std::string m_title_id = psf::get_string(insdir_psf, "TITLE_ID");
+	std::string local_path = Emu.GetHddDir() + "game/" + m_title_id;
+
+	// If dir not found on hdd0 , it's a first install and we can install any install packages
+	if (!fs::is_dir(local_path))
+	{
+		//Insdir check. We need to install the packages over the game install if this is the case.
+		const std::string insdir_disk_dir = path + "/PS3_GAME/INSDIR";
```

Comment:
I would suggest to split processing of each directory in the separate method to keep them smaller and more maintainable.